ansys/dpf/core/server.py

```diff
@@ -351,25 +354,31 @@ def __init__(
         check_valid_ip(ip)
         if not isinstance(port, int):
             raise ValueError("Port must be an integer")
+        address = "%s:%d" % (ip, port)
 
         if os.name == "posix" and "ubuntu" in platform.platform().lower():
             raise OSError("DPF does not support Ubuntu")
         elif launch_server:
-            self._server_id = launch_dpf(str(ansys_path), ip, port,
-                                         docker_name=docker_name,
-                                         timeout=timeout)
+            if is_pypim_configured() and not ansys_path and not docker_name:
+                self._remote_instance = launch_remote_dpf()
+                address = self._remote_instance.services["grpc"].uri
+            else:
+                ip = ip or LOCALHOST
+                port = port or DPF_DEFAULT_PORT
+                self._server_id = launch_dpf(str(ansys_path), ip, port,
+                                            docker_name=docker_name,
+                                            timeout=timeout)
 
-        self.channel = grpc.insecure_channel("%s:%d" % (ip, port))
+        self.channel = grpc.insecure_channel(address)
 
         # assign to global channel when requested
         if as_global:
             dpf.core.SERVER = self
 
         # TODO: add to PIDs ...
 
-        # store port and ip for later reference
-        self._input_ip = ip
```

Comment:
@plule-ansys would it eventually be possible to keep self._input_ip and self._input_port at None ?

Reply:
Sure done. I'm keeping them as-is in the general case, and setting them to "None" when an address is provided.